src/providers/VoiceFocusProvider/index.tsx

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  useMemo,
+  useContext,
+  createContext,
+  useCallback,
+} from 'react';
+import {
+  Device,
+  VoiceFocusTransformDevice,
+  VoiceFocusDeviceTransformer,
+  VoiceFocusSpec,
+  VoiceFocusDeviceOptions,
+} from 'amazon-chime-sdk-js';
+
+interface Props {
+  /** Determines how you want the Amazon Voice Focus to behave. This spec is used to derive a runtime configuration when a transformer is created. */
+  spec?: VoiceFocusSpec;
+  /**
+  * A set of options that can be supplied when creating an Amazon Voice Focus device.
+  * For more info, you can go to https://aws.github.io/amazon-chime-sdk-js/interfaces/voicefocusdeviceoptions.html
+  */
+  options?: VoiceFocusDeviceOptions;
+}
+
+interface VoiceFocusState {
+  isVoiceFocusSupported: boolean | undefined;
+  addVoiceFocus: (device: Device) => Promise<Device | VoiceFocusTransformDevice>;
+}
+
+const VoiceFocusContext = createContext<VoiceFocusState | null>(null);
+
+const VoiceFocusProvider: React.FC<Props> = ({
+  spec,
+  options,
+  children
+}) => {
+  const [isVoiceFocusSupported, setIsVoiceFocusSupported] = useState<boolean | undefined>(undefined);
+  const [voiceFocusDevice, setVoiceFocusDevice] = useState<VoiceFocusTransformDevice | null>(null);
+  const [voiceFocusTransformer, setVoiceFocusTransformer] = useState<VoiceFocusDeviceTransformer | null>(null);
+
+  // Check whether the value of spec is undefined or empty object
+  const vfSpec = useMemoCompare(spec, (prev: any, next: any) => {
+    const FIXED_EMPTY = {};
+    next === undefined ? FIXED_EMPTY : Object.keys(next).length === 0 ? FIXED_EMPTY : next;
+    return prev && Object.keys(prev).length === 0 && Object.keys(next).length === 0;
+  });
+
+  function useMemoCompare(next: any, compare: any) {
```

Comment:
You should move as much shared logic as you can out of this component and into a helper file somewhere.

Reply:
Thanks for it, removed this part to the `util` folder